pkg/apis/core/validation/utils.go

```diff
@@ -134,3 +134,17 @@ func equal(new, old []string) bool {
 	}
 	return true
 }
+
+func isIgnoreTaintsUnique(ignoredTaints []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	taintKeySet := make(map[string]struct{})
+
+	for index, taint := range ignoredTaints {
+		if _, ok := taintKeySet[taint]; ok {
+			allErrs = append(allErrs, field.Duplicate(fldPath.Index(index), taint))
+			continue
+		}
+		taintKeySet[taint] = struct{}{}
+	}
+	return allErrs
+}
```

Comment:
You could move this to `pkg/apis/core/validation/shoot.go`. Currently I don't see benefit/reason why we would want to keep it in this utils file.